homeassistant/components/nextbus/__init__.py

```diff
@@ -4,15 +4,41 @@
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
 
+from .const import CONF_AGENCY, CONF_ROUTE, CONF_STOP, DOMAIN
+from .coordinator import NextBusDataUpdateCoordinator
+
 PLATFORMS = [Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up platforms for NextBus."""
+    entry_agency = entry.data[CONF_AGENCY]
+
+    coordinator: NextBusDataUpdateCoordinator = hass.data.setdefault(DOMAIN, {}).get(
+        entry_agency
+    )
+    if coordinator is None:
+        coordinator = NextBusDataUpdateCoordinator(hass, entry_agency)
+        hass.data[DOMAIN][entry_agency] = coordinator
+
+    coordinator.add_stop_route(entry.data[CONF_STOP], entry.data[CONF_ROUTE])
+
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    await coordinator.async_config_entry_first_refresh()
+
```

Comment:
```suggestion
```

Reply:
Why did mark this conversation resolved without implementing the requested change?
`await coordinator.async_config_entry_first_refresh()` should be before `await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)`

Reply:
Oh, I had done this before, but reverted it. This breaks all tests because the state is undefined.

This is because the data in the sensors is set as part of `_handle_coordinator_update()`. By doing the refresh prior to adding there is no sensor to "handle" the update. Later, when the sensor is added, it does not know to fetch content.

There is possibly (probably?) some additional hook that could be used. However, is there a reason why the current arrangement is bad practice? It seems like the end result is the same.

Reply:
I found a way to do this. I'm still curious on the reasoning. One thing I came up with is that doing the first refresh first would allow catching configuration errors before sensors and such get added. I don't remember seeing this recommended in the documentation though.

Reply:
I will check the docs later. 
The reason to call ` coordinator.async_config_entry_first_refresh()` first is simple that the integration should return quickly on errors.
For example, it makes no sense to set up all entities when there is a connection problem with the API.